static_analyzer/analysis_cache.py

```
@@ -264,30 +261,22 @@ def save(self, result: "StaticAnalysisResults", source_sha: str | None = None) -
-            elif self.sha_path.exists():
-                # No SHA provided this run; drop any stale tag so the next
-                # SHA-gated read doesn't accidentally accept a mismatched pickle.
+            tag_text = f"{_TAG_VERSION}\n{source_sha}\n"
+            tag_fd, tag_tmp = tempfile.mkstemp(dir=self.artifact_dir, suffix=".sha.tmp")
+            try:
+                with open(tag_fd, "w", encoding="utf-8", newline="\n") as f:
+                    f.write(tag_text)
+                    f.flush()
+                    os.fsync(f.fileno())
+                Path(tag_tmp).replace(self.sha_path)
+            except Exception as e:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)